x-pack/solutions/observability/plugins/infra/server/lib/alerting/inventory_metric_threshold/register_inventory_metric_threshold_rule_type.ts

```diff
@@ -123,6 +81,11 @@ export function registerInventoryThresholdRuleType(
     producer: 'infrastructure',
     minimumLicenseRequired: 'basic',
     isExportable: true,
+    /*
+     * The schema defined in response-ops-rule-params cannot import all types from the plugins
+     * so the executor will expect slight differences with the type InventoryMetricThresholdParams.
+     */
+    // @ts-ignore
```

Comment:
This was inevitable without a major refactor of the types in `x-pack/solutions/observability/plugins/infra/common/alerting/metrics/types.ts`.

I tried changing `InventoryMetricThresholdParams` to `TypeOf<typeof metricInventoryThresholdRuleParamsSchema>;` but it was not possible without a major refactor.

The schema is not seen as exactly the same as `InventoryMetricThresholdParams` because the response-ops-rule-params package cannot import from `'@kbn/metrics-data-access-plugin/common'` so some types are mismatched(although they are technically the same.